#!/usr/bin/env python

PACKAGE = "lidar_team_erp42"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("small_st_minPoints", int_t, 0, "small_st_minPoints", 30, 1, 100)
gen.add("small_st_epsilon", double_t, 0, "small_st_epsilon", 0.4, 0, 20)
gen.add("small_st_minClusterSize", int_t, 0, "small_st_minClusterSize", 10, 1, 100)
gen.add("small_st_maxClusterSize", double_t, 0, "small_st_maxClusterSize", 5000, 1, 100000)


#xMaxROI = 10
gen.add("small_st_xMinROI", double_t, 0, "small_st_xMinROI", 0.5, -200, 10)
gen.add("small_st_xMaxROI", double_t, 0, "small_st_xMaxROI", 5.0, -10, 200)
gen.add("small_st_yMinROI", double_t, 0, "small_st_yMinROI", -1.0, -200, 10)
gen.add("small_st_yMaxROI", double_t, 0, "small_st_yMaxROI", 1.0, -10, 200)
gen.add("small_st_zMinROI", double_t, 0, "small_st_zMinROI", -0.40, -2, 0)
gen.add("small_st_zMaxROI", double_t, 0, "small_st_zMaxROI", 3.0, -2, 100)

gen.add("small_st_xMinBoundingBox", double_t, 0, "small_st_xMinBoundingBox", 0.2, 0, 10);
gen.add("small_st_xMaxBoundingBox", double_t, 0, "small_st_xMaxBoundingBox", 2.0, 0, 10);
gen.add("small_st_yMinBoundingBox", double_t, 0, "small_st_yMinBoundingBox", 0.2, 0, 10);
gen.add("small_st_yMaxBoundingBox", double_t, 0, "small_st_yMaxBoundingBox", 2.0, 0, 10);
gen.add("small_st_zMinBoundingBox", double_t, 0, "small_st_zMinBoundingBox", 0.1, 0, 0.1);
gen.add("small_st_zMaxBoundingBox", double_t, 0, "small_st_zMaxBoundingBox", 2.0, 0, 10);

exit(gen.generate(PACKAGE, "small_static_obstacle", "small_st_hyper_parameter"))
